src/Symfony/Component/HttpKernel/Tests/HttpClientKernelTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpClientKernel;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class HttpClientKernelTest extends TestCase
+{
+    public function testHandlePassesMaxRedirectsHttpClientOption()
+    {
+        $request = new Request();
+        $request->attributes->set('http_client_options', ['max_redirects' => 50]);
+
+        $response = $this->getMockBuilder(ResponseInterface::class)->getMock();
+        $response->expects($this->once())->method('getStatusCode')->willReturn(200);
+
+        $client = $this->getMockBuilder(HttpClientInterface::class)->getMock();
```

Comment:
in order to use these classes, you need to list symfony/http-client in the composer.json file of HttpKernel (this should fix tests, they fail currently)

Reply:
Thanks for pointing it out. First time that I contribute to symfony. :-)

Reply:
I only added `symfony/http-client-contracts` as the real implementation isn't required for the test.